tests/parsers/test.manifestjson.js

```diff
@@ -279,3 +279,57 @@ describe('ManifestJSONParser update_url', function() {
     assert.equal(manifestJSONParser.isValid, true);
   });
 });
+
+
+describe('ManifestJSONParser with comments', function() {
+
+  it('parses JSON', () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = `// I am a JSON comment, sigh\n${validManifestJSON()}`;
+    var manifestJSONParser = new ManifestJSONParser(json,
+                                                    addonLinter.collector);
+    assert.equal(manifestJSONParser.isValid, true);
+  });
+
+  it('parses the example from Chrome developer docs', () => {
+    var addonLinter = new Linter({_: ['bar']});
+    // Example from https://developer.chrome.com/extensions/manifest
+    var json = [
+      '{',
+      '// Required',
+      '"manifest_version": 2,',
+      '"name": "My Extension",',
+      '// Make the hell sure to use semvar.org if increasing this',
+      '"version": "0.0.1"',
+      '}',
+    ].join('\n');
+    var manifestJSONParser = new ManifestJSONParser(json,
+                                                    addonLinter.collector);
+    assert.equal(manifestJSONParser.isValid, true);
+    assert.notInclude(manifestJSONParser._jsonString, 'semvar.org');
+  });
+
+  it('returns the correct error for malformed JSON', () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = `{"something": true,\n// I am a JSON comment, sigh\nblah}`;
+    var manifestJSONParser = new ManifestJSONParser(json,
+                                                    addonLinter.collector);
+    assert.equal(manifestJSONParser.isValid, false);
+    var errors = addonLinter.collector.errors;
+    assert.equal(errors[0].code, messages.MANIFEST_JSON_INVALID.code);
+    assert.include(errors[0].message, 'Invalid JSON in manifest file.');
+  });
+
+  it("doesn't evaluate JS code even though esprima is used", () => {
+    var addonLinter = new Linter({_: ['bar']});
+    var json = '// eval("");\n{"something": true}\nvar bla = "foo";';
```

Comment:
and… `'{"something": eval("")}'` is invalid too? I would assume that `eval()` in a comment doesn't work but outside could be a more realistic test maybe